test/e2e/e2e.go

```diff
@@ -211,58 +214,176 @@ var _ = ginkgo.Describe("[efs-csi] EFS CSI", func() {
+	podList, err := f.ClientSet.CoreV1().Pods("kube-system").List(
+		metav1.ListOptions{
+			LabelSelector: labelSelector,
+			FieldSelector: fieldSelector,
+		})
+	framework.ExpectNoError(err, "getting driver pod")
+	framework.ExpectEqual(len(podList.Items), 1, "expected 1 efs csi node pod but got %d", len(podList.Items))
+	driverPod := podList.Items[0]
+
+	ginkgo.By(fmt.Sprintf("Deleting driver pod %q on node %q", driverPod.Name, node.Name))
+	err = e2epod.DeletePodWithWaitByName(f.ClientSet, driverPod.Name, "kube-system")
+	framework.ExpectNoError(err, "deleting driver pod")
+}
+
+func createPvPvcWithPv(f *framework.Framework, pv *v1.PersistentVolume) (*v1.PersistentVolume, *v1.PersistentVolumeClaim) {
+	pvc, pv := makePVCWithPV(pv, f.Namespace.Name, f.Namespace.Name)
+	pvc, pv, err := createEFSPVPVC(f.ClientSet, f.Namespace.Name, pvc, pv)
+	framework.ExpectNoError(err, "creating efs pvc & pv")
+
+	return pv, pvc
+}
+
+func validatePodVolumeAccessible(f *framework.Framework, pod *v1.Pod) {
+	ginkgo.By(fmt.Sprintf("Execing a write to the pod %q in namespace %q", pod.Name, pod.Namespace))
+	command := fmt.Sprintf("touch /mnt/volume1/%s-%s", f.Namespace.Name, time.Now().Format(time.RFC3339))
+	done := make(chan bool)
+	go func() {
+		defer ginkgo.GinkgoRecover()
+		utils.VerifyExecInPodSucceed(f, pod, command)
+		done <- true
+	}()
+	select {
+	case <-done:
+		framework.Logf("verified exec in pod succeeded")
+	case <-time.After(30 * time.Second):
+		framework.Failf("timed out verifying exec in pod succeeded")
+	}
+}
+
+func launchPod(f *framework.Framework, pv *v1.PersistentVolume, pvc *v1.PersistentVolumeClaim) (*v1.Node, *v1.Pod) {
+	node, err := e2enode.GetRandomReadySchedulableNode(f.ClientSet)
+	framework.ExpectNoError(err, "getting random ready schedulable node")
+	command := fmt.Sprintf("touch /mnt/volume1/%s-%s && trap exit TERM; while true; do sleep 1; done", f.Namespace.Name, time.Now().Format(time.RFC3339))
+
+	isPrivileged := false
+	var nodeSelector map[string]string
+	pod := e2epod.MakePod(f.Namespace.Name, nodeSelector, []*v1.PersistentVolumeClaim{pvc}, isPrivileged, command)
+	ginkgo.By(fmt.Sprintf("Creating pod on node %q to mount pvc %q at namespace %q, pv %q and run %q", node.Name, pvc.Name, f.Namespace.Name, pv.Name, command))
+	pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
+	framework.ExpectNoError(err, "creating pod")
+	framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, pod.Name, f.Namespace.Name), "waiting for pod running")
+
+	return node, pod
+}
+
+func updateDriverTo(daemonSetClient clientgoappsv1.DaemonSetInterface, driverUpgradeTo *appsv1.DaemonSet) {
+	ginkgo.By("Updating to the EFS CSI driver")
+	currentEfsDriver, err := daemonSetClient.Get("efs-csi-node", metav1.GetOptions{})
+	driverUpgradeTo.ResourceVersion = currentEfsDriver.ResourceVersion
+	_, err = daemonSetClient.Update(driverUpgradeTo)
```

Comment:
Need to wait for daemonset to be totally replaced. Or at least the driver pod on the node we care about